action.yml

```diff
@@ -41,7 +46,7 @@ runs:
   using: 'node20'
   main: 'dist/restore/index.js'
   post: 'dist/save/index.js'
-  post-if: "success()"
+  post-if: "success() && github.event.inputs.save-on-success"
```

Comment:
This will not work for the same reason that `save-always` was removed - #1452.

`post-if` does not have access to the input context, so this will always be false and completely break saving the cache.

To enable this type of workflow, you need to either:
1. Enable accessing the input context in `post-if` (requires `actions/runner` changes)
2. Use two separate actions for restore and save (this should already work)